### ISSUE TITLE
Head script caching issue

### ISSUE THREAD

**Issue by lscorcia** [Fabrik/fabrik#2282]
Hi, we are using Fabrik to display some tables on a public-facing website. We noticed that when bots come index our site, the number of sessions ballons (not a Fabrik issue), but we also noticed that those sessions tend to become larger and larger, up to several megabytes, effectively DDOSsing our session storage.
We decoded a few of those session vars and we noticed that the bulk of their size is due to the Fabrik head script cache, i.e.:

https://github.com/Fabrik/fabrik/blob/master/plugins/system/fabrik/fabrik.php#L217-L251

The issue is that there is no upper bound on the number of URLs whose scripts will be cached. Ideally, one should keep just a handful of cache entries around in session, the latest 10 or 20 would be plenty, but in our codebase we easily reach 900+ URLs per session!

The problem is heavily visible when bots show up as they do not throttle their requests like a human would do and can easily submit thousands of requests in a single session.

Any idea about how to fix this problem?
Thanks in advance

**Comment by trob** [comment]
This is Fabrik 3 for Joomla!3 which will reach its end-of-life in 3 days.

**Comment by lscorcia** [comment]
Yep, I know. Unfortunately we'll have to keep Joomla 3 for the foreseeable future (not my choice). Would Fabrik for J!4 be affected by this issue as of now?

**Comment by trob** [comment]
I can't see any code change in F4 in this case
